src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -117,7 +117,7 @@ internal sealed class ContainedDocument : ForegroundThreadAffinitizedObject, IVi
 
             this.Key = new DocumentKey(Project, filePath);
             this.Id = DocumentId.CreateNewId(Project.Id, filePath);
-            this.Folders = containedLanguage.Project.GetFolderNames(itemId);
+            this.Folders = (containedLanguage.Project as AbstractLegacyProject)?.GetFolderNames(itemId);
```

Comment:
Are you trying to support ContainedDocuments in non-legacy scenarios? Otherwise I think I'd prefer a hard cast; this might just flow strange data around and it'll be hard to debug.